lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -975,24 +975,30 @@ public function getName()
      */
     protected function doModifyLimitQuery($query, $limit, $offset = null)
     {
-        $limit = (int) $limit;
```

Comment:
@deeky666 but unlike `OFFSET`, for `LIMIT` `0` and `null` have different meanings — "0 records" and "all records" respectively. There's no complete consistency between existing adapters but all of them except SQL Anywhere only omit `$limit` when it's `null`. Technically, it is possible to execute a query and retrieve 0 records from it, so the DBAL shouldn't be a limiting factor here. Also from the API consumer standpoint, if there's a bug in the calculation of `$limit` on the application end and it's calculated as `0`, I'd better expect 0 records to be returned instead of all of them.